metrics/fidelity/utility.py

```diff
@@ -0,0 +1,174 @@
+    """
+    clip=False ERASER's definition of sufficiency and comprehensiveness Cite?? ask chenhao
+    Calculating the fidelity value given probabilities
+    :param prob_y_hat: P(y_hat/x)
+    :param prob_y_hat_alpha: P(y_hat/x,alpha) where text is reduced by alpha
+    :param fidelity_type: sufficiency or comprehensiveness
+    :param clip: true for clipping the fidelity values between 0 and 1
+    :param dataset_level: dataset level fidelity else instance level
+    :return: fidelity
+    """
+    if clip:
+        mean_difference = np.clip(np.array(prob_y_hat) - np.array(prob_y_hat_alpha), 0, 1)
+    else:
+        mean_difference = prob_y_hat - prob_y_hat_alpha
+    if dataset_level:
+        mean_difference = np.mean(mean_difference)
+    if fidelity_type == "sufficiency":
+        return 1 - mean_difference
+    else:
+        return mean_difference
+
+
+def itemwize_normalization(fidelity, null_difference, fidelity_type, clip, eps):
+    """
+    Normalize a fidelity value given the null difference value
+    :param fidelity: sufficiency or comprehensiveness metric
+    :param null_difference: max(0, P(y_hat/x) - P(y_hat/x, 0)); alpha = 0
+    :param fidelity_type: sufficiency or comprehensiveness
+    :param clip: true for clipping the fidelity values between 0 and 1
+    :param eps: denominator threshold when it tends to 0
+    :return: normalized fidelity
+    """
+    try:
+        if fidelity_type == "sufficiency":
+            numerator = fidelity - (1 - null_difference)
+            denominator = 1 - (1 - null_difference)
+        else:
+            numerator = fidelity
+            denominator = null_difference
+
+        if np.abs(denominator) < eps:
+            result = 0
+        else:
+            result = numerator / denominator
+    except Exception as e:
+        logger.info(e)
+        return 0
+
+    if clip:
+        return np.clip(result, 0, 1)
+    else:
+        return result
+
+
+def normalization(fidelity, null_difference, fidelity_type="sufficiency", clip=True, eps=1e-5):
+    """
+    Normalize a list of fidelity values given the null difference values
+    :param fidelity: sufficiency or comprehensiveness metric
+    :param null_difference: max(0, P(y_hat/x) - P(y_hat/x, 0)); alpha = 0
+    :param fidelity_type: sufficiency or comprehensiveness
+    :param clip: true for clipping the fidelity values between 0 and 1
+    :param eps: denominator threshold when it tends to 0
+    :return: normalized fidelity
+    """
+    normalized_fidelity = []
+    for idx, itemwize_fidelity in enumerate(fidelity):
+        normalized_fidelity.append(
+            itemwize_normalization(
+                fidelity=itemwize_fidelity,
+                null_difference=null_difference[idx],
+                fidelity_type=fidelity_type,
+                clip=clip,
+                eps=eps,
+            )
+        )
+
+    return normalized_fidelity
+
+
+def compute_predictions(input_ids, model, attention_masks=None):
```

Comment:
Do type hinting. also use consistent terminology. What you are calling "attention_masks" in this function, I believe you are calling "alpha" above. Pick one term and stick with it.